File: SoftUni - Advanced/Iterators and Comparators/Iterators and Comparators - Exercise/6.Equality_Logic/Person.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _6.Equality_Logic
{
    public class Person : IComparable<Person>
    {
        private string name;

        private int age;

        public Person(string name, int age)
        {
            this.name = name;
            this.age = age;
        }

        public string Name => name;

        public int Age => age;

        public int CompareTo(Person other)
        {
            int result = Name.CompareTo(other.Name);
            if (result == 0)
            {
                result = Age.CompareTo(other.Age);
            }
            return result;
        }

        public override int GetHashCode()
        {
            return Name.GetHashCode() ^ Age.GetHashCode();
        }

        public override bool Equals(object obj)
        {
            var other = obj as Person;
            if (other == null)
            {
                return false;
            }

            return Age == other.Age && Name == other.Name;
        }

        public override string ToString()
        {
            return $"{Name} {Age}";
        }
    }
}

File: SoftUni - Java Advanced/Iterators and Comparators/Iterators and Comparators - Exercises/_05_Comparing_Objects/Person.java
package _05_Comparing_Objects;

public class Person implements Comparable<Person> {
    private String name;
    private int age;
    private String town;

    public Person(String name, int age, String town) {
        this.name = name;
        this.age = age;
        this.town = town;
    }

    @Override
    public int compareTo(Person person) {
        int result = 0;

        if(this.name.equals(person.name)) {
            if(this.age == person.age) {
               return this.town.compareTo(person.town);
            }
        }
        return this.name.compareTo(person.name);
    }
}
